src/BibleWell.App/ViewModels/MenuItemTemplate.cs

```diff
@@ -1,15 +1,31 @@
 ﻿using BibleWell.App.ViewModels.Pages;
+using CommunityToolkit.Mvvm.ComponentModel;
 
 namespace BibleWell.App.ViewModels;
 
-public sealed class MenuItemTemplate(Type _type, string _iconName)
+public partial class MenuItemTemplate(Type viewModelType, string iconName, bool isSvg = false, string? iconNameInactive = null)
+    : ObservableObject
 {
-    public string IconName { get; } = _iconName;
+
+    [ObservableProperty]
+    private bool _isSelected;
 
     // TODO localization
-    public string Label { get; } = _type.Name.Replace("PageViewModel", "");
+    public string Label { get; } = viewModelType.Name.Replace("PageViewModel", "");
+
+    public Type ViewModelType { get; } = viewModelType.BaseType == typeof(PageViewModelBase)
+        ? viewModelType
+        : throw new ArgumentException($"Type must derive from {nameof(PageViewModelBase)}.", nameof(viewModelType));
+    public string IconName { get; } = iconName;
+
+    public string? IconNameInactive { get; } = iconNameInactive;
```

Comment:
Should this fall back to the icon name when there isn't an inactive icon?
```suggestion
    public string? IconNameInactive { get; } = iconNameInactive ?? iconName;
```